### ISSUE TITLE
Could `intersection_candidates` only require the envelopes to be the same?

### ISSUE THREAD

**Issue by Andlon** [Stoeoef/rstar#23]
First, thank you very much for this crate, it has worked very well for me so far. I like the API and the flexibility, and of course the performance!

I have a case where I want to determine intersection candidates between triangles and tetrahedra. I have two separate RTrees for other purposes (e.g. finding closest triangle/tetrahedron to a point): one for triangles and one for tetrahedra. Unfortunately, `intersection_candidates_with_other_tree` only accepts `&Self` as input, despite the fact that it only works with the envelopes of the two objects, which are the same (`AABB`). Would it be possible to relax `intersection_candidates_with_other_tree` so that it takes any `other` which shares the same envelope?

**Comment by Stoeoef** [comment]
That's a good observation. I don't see a reason why this shouldn't be possible.

The implementation should also be straightforward: Change the type signature like this:

```Rust
    pub fn intersection_candidates_with_other_tree<'a, U>(
        &'a self,
        other: &'a RTree<U>,
    ) -> IntersectionIterator<T, U>
    where
        U: RTreeObject<Envelope = T::Envelope>,
```

and then continue

**Comment by Andlon** [comment]
@Stoeoef: indeed, it was quite straightforward. See my PR :-) Also note the point about ordering of the returned tuples.

**Comment by Stoeoef** [comment]
Implemented with v0.7.1
